Helper/Cart.php

```diff
@@ -1363,6 +1363,15 @@ function ($item) use ($imageHelper, &$totalAmount, &$diff, $storeId, $currencyCo
             $products[] = $product;
         }
 
+        if ($this->deciderHelper->isHandleVirtualProductsAsPhysical()) {
```

Comment:
You could have added this condition in the initial loop, to the line 1278 above.